respect-app-compose/src/commonMain/kotlin/world/respect/app/viewmodel/CurriculumEditViewModel.kt

```diff
@@ -0,0 +1,208 @@
+
+package world.respect.app.viewmodel
+
+import androidx.lifecycle.SavedStateHandle
+import kotlinx.coroutines.flow.*
+import kotlinx.coroutines.launch
+import world.respect.shared.viewmodel.RespectViewModel
+import world.respect.app.domain.models.Curriculum
+import androidx.lifecycle.viewModelScope
+import world.respect.shared.navigation.NavCommand
+import world.respect.shared.viewmodel.app.appstate.AppUiState
+import world.respect.shared.navigation.CurriculumDetail
+import world.respect.shared.navigation.CurriculumList
+import world.respect.app.domain.usecase.curriculum.GetCurriculumByIdUseCase
+import world.respect.app.domain.usecase.curriculum.SaveCurriculumUseCase
+
+class CurriculumEditViewModel(
+    savedStateHandle: SavedStateHandle,
+    private val getCurriculumByIdUseCase: GetCurriculumByIdUseCase,
+    private val saveCurriculumUseCase: SaveCurriculumUseCase
+) : RespectViewModel(savedStateHandle) {
+
+    private val curriculumId: String? = savedStateHandle.get<String>("curriculumId")
+
+    data class CurriculumEditUiState(
+        val curriculum: Curriculum? = null,
```

Comment:
Having a curriculum entity object on the uistate, in addition to name, id, and description is violation of single point of truth: creates ambiguity. Should use only the Curriculum entity.